src/App.tsx

```diff
@@ -1,10 +1,13 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, MouseEvent } from 'react';
 import './App.css';
+import Nav from './Nav';
 import CardGrid from './CardGrid';
+import DetailPage from './DetailPage';
 
 const App: React.FC = () => {
 
   const [allPokemon, setAllPokemon] = useState([]);
+  const [currentPokemon, setCurrentPokemon] = useState(null);
```

Comment:
Something like `useState<Pokemon | null>(null);` should work.

But before you get too far down this road, the "current pokemon" should be determined by what route (i.e. `/150`) we're on. Think about linking directly to a detail page. We'll need to read the id from the route and load the info we need.

Reply:
"We'll need to read the id from the route and load the info we need." Love it! I can do that in vanilla React, but I'm stuck trying to execute it in Typescript. I don't know how to get Typescript to accept the fact that I'm trying to draw on both component props and resource parameters.